### ISSUE TITLE
[Crash]: AI Controlled Rockets

### ISSUE THREAD

**Issue by DarkEviL88** [Sgt-Imalas/Sgt_Imalas-Oni-Mods#70]
### Description of the bug/crash

NullReferenceException

UnityEngine.Component.GetComponent[T] () (at <72b60a3dd8cd4f12a155b761a1af9144>:0)
KPrefabIDExtensions.HasTag (UnityEngine.Component cmp, Tag tag) (at <cd4b9bd5aa6c4ec38ec00dca1dc79105>:0)
RoboRockets.RoboRocketPatches+DisableViewInteriorWorldSelector_Patch.Prefix (System.Int32 id) (at <a2097e5e2d0f4305a921b6f984e894dc>:0)
(wrapper dynamic-method) WorldSelector.WorldSelector.OnWorldRowClicked_Patch1(WorldSelector,int)
WorldSelector+<>c__DisplayClass11_0.<SpawnToggles>b__0 () (at <cd4b9bd5aa6c4ec38ec00dca1dc79105>:0)
MultiToggle.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at <cd4b9bd5aa6c4ec38ec00dca1dc79105>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at <e9635660f69b4d9d84c085aeda3be353>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at <e9635660f69b4d9d84c085aeda3be353>:0)
UnityEngine.EventSystems.EventSystem:Update()

Build: U47-562984-S

### Steps To Reproduce

This rarely happens, and if I now launch the game and follow the same steps, the game will not crash. I just clicked on a rocket with an AI brain in the list of asteroids and the game crashed, I have a mod installed to rename asteroids, it allows you to rename rockets too, but after saving and loading the game, the names of the rockets return to the previous ones, perhaps some kind of conflict with it. Today I played ONI for a very long time and clicked from the star map on a rocket with an AI brain and from other asteroids and the game did not crash, but after a long time it crashed. In general, this happens randomly when you click on the rocket with AI. The crash occurs exactly when clicking on the rocket with AI brain in the list of planetoids. I don't know if it will crash using AI Control Module and AI Nosecone.

### Player.log

[Player.log](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/files/12113734/Player.log)

**Comment by Sgt-Imalas** [comment]
renaming rockets is built into the game, so that might be that renaming mod bugging out.
the crash sounds like a race condition, I'll try to mitigate that

**Comment by DarkEviL88** [comment]
Yeah, I noticed that the crash usually happened after renaming a rocket in the planetoids list using the rename asteroids mod. I renamed the rockets in a different menu and there are no crashes and their new names are finally saved. Now I wonder what mod bugs rovers, sometimes they have broken animation, sometimes it start jumping up and down on the ladder and can not go up. And the second rover bug when it run out of charge, it continue to work until you reload the save.

**Comment by Sgt-Imalas** [comment]
the rovers jumping is a base game bug that happens if they get displaced for some reason (f.e. meteors did that until recently).
the "rovers not dying" is caused by the "dying" being a task with priority 5, and sometimes the rover just has better stuff to do

**Comment by DarkEviL88** [comment]
Perhaps we should close the issue, I have not noticed any more game crashes. I apologize for being off topic. Found a bug that if you build pipes (for gas or liquid) on the edge of the map in the shadows, the flow of liquid or gas stops flowing, and if you cut the pipe with scissors or demolish the command module with a connected pipe, the game crashes. Installed your mod Rocketry Expanded, I haven't looked at all the content yet, but the mod is awesome. I translated it into Russian and now I'm checking how it looks in the game, as soon as I finish I can share the translation if needed.

**Comment by Sgt-Imalas** [comment]
I'd recommend holding translating RE for now, the mod is (unofficially) still in beta, as the whole space station system, explorable derelics, route planner tool and the whole space station science systems (among other things) are still missing, so there will be a lot of new strings that aren't there yet

**Comment by DarkEviL88** [comment]
So that's why I didn't find the buildings in the game that I saw in the translation. Well for now I will leave the translation for myself perhaps improve it while the mod is in beta

**Comment by DarkEviL88** [comment]
It is strange that neutronium alloy does not give increased temperature of overheating for buildings, the alloy is made of steel, which increases the temperature of overheating +200, and since neutronium has no thermal conductivity, buildings should not warm up at all, I tried to build a pump and pump out the liquid metal, but the pump overheated. From the description I understood that neutronium alloy is only suitable for building buildings so far, but not mechanisms. When I first got Insulation, I was expecting a higher overheating temperature from Thermium so I could pump super-hot liquids, but I was very disappointed. The developers of the game made 2 coolest things - Insulation and Super Coolant, but 3 - Thermium screwed up a lot as for me, I want to have the coolest alloy at the end of the game to build pumps and they did not overheat from super-hot liquids, I do not know all the chips of the game, but in most cases is enough just steel and for Niobium I no longer fly. I still haven't learned how to properly use the Turbine and Thermo Aquatuner combo.

**Comment by Sgt-Imalas** [comment]
Very good point

**Comment by DarkEviL88** [comment]
On large tanks with liquid fuel the icon when there is no pipe connected has disappeared ![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/8fd1b0a6-deca-4045-89a7-61ca6269a598), but this line has appeared, too lazy to specify, but I think it's because of RE.
![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/81fb7452-e30c-4b67-87d4-8bb6d38b4898)

**Comment by DarkEviL88** [comment]
![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/b8c3d507-a64b-4643-a272-1aa9eaf70e7f)
![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/68cdc60f-cfc4-4444-9a43-daa353a2c64e)
In these two command modules the fps dropped to 30 from 50+, maybe I had to restart the game, but in the standard command modules with fps everything is fine, also everything is fine in these two modules
![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/5a22c845-12a8-4ef4-8f91-1510cc20af1c)
![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/3c9c00b0-4a57-4a59-8a38-79bca3db9ea2)
I apologize for the screenshots and not the full text name, the mods are translated and I don't remember it's English names.

**Comment by Sgt-Imalas** [comment]
the fps dropping might be fixed now.
I never had it happen to me but it might have been related to the slightly buggy code in the camera controller that could cause flickering inside these smaller modules.
I have replaced that code in 2.7

**Comment by DarkEviL88** [comment]
I was wrong to write that neutronium alloy should not conduct heat, pure neutronium itself does not conduct heat, but it is an alloy with steel and I think neutronium alloy should conduct some heat. I faced the problem that when building Thermo Aquatuner of neutronium alloy, it can not be cooled, I built it in the water to heat it up to the state of steam and continued to heat up the sensor to 200 degrees to further utilize in the turbine, but Thermo Aquatuner does not heat the water, I have installed a mod for [Thermal Interface Plate](https://steamcommunity.com/sharedfiles/filedetails/?id=1938747693) with it part of the heat from Thermo Aquatuner goes into the atmosphere, but the water is not heated as effectively as with Thermo Aquatuner of steel. The Thermo Aquatuner constructed of neutronium alloy with Thermal Interface Plate in the water simply cools down giving up minimal heat to the water. Tempshift Plate doesn't seem to work at all. I'd like to keep the 2000 degree overheat temp, but add some % of thermal conductivity. Just how to balance the thermal conductivity of this alloy, 1\3 thermal conductivity from steel? It seems to me that there should be thermal conductivity, but not too much. And 100% radiation protection can be left as is. Judge for yourself, if 0 thermal conductivity was your idea and you don't want to change it, I don't want to interfere and impose my opinion.

**Comment by DarkEviL88** [comment]
In the neutronium alloy fabrication database I found this line, maybe I didn't translate something or used an old template.
![image](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/assets/50534207/1de86b15-9979-4087-b384-86435c174e69)


Although in the description it says "Insulator" and it has 0 thermal conductivity, and then I thought about the fact that I wrote all the text above for nothing. I want the game to have mechanisms with a very high temperature of overheating, but also to cool it down was possible, but on the other hand it will break your idea of insulating alloy.
